.travis.yml

```diff
@@ -1,16 +1,39 @@
 language: go
-go:
-- 1.8
 sudo: false
-install:
-- go get -t ./...
-- go get -u github.com/golang/lint/golint
-- go get -u golang.org/x/tools/cmd/goimports
-script:
-- go test -v $(go list ./... | grep -v /vendor/)
-- go vet $(go list ./... | grep -v /vendor/)
-- diff <(goimports -d $(find . -type f -name '*.go' -not -path "./vendor/*")) <(printf "")
-- (for d in $(go list ./... | grep -v /vendor/); do diff <(golint $d) <(printf "") || exit 1;  done)
+matrix:
+  include:
+  # Using vendored dependencies
+  - install:
+    - go get -u github.com/golang/dep/...
+    - go get -u github.com/golang/lint/...
+    - go get -u golang.org/x/tools/cmd/...
+    script:
+    - dep ensure
+    - go test -v $(go list ./... | grep -v /vendor/)
+    - go vet $(go list ./... | grep -v /vendor/)
+    - diff <(goimports -d $(find . -type f -name '*.go' -not -path "./vendor/*")) <(printf "")
+    - (for d in $(go list ./... | grep -v /vendor/); do diff <(golint $d) <(printf "") || exit 1;  done)
+  # Using HEAD of dependencies
+  - install:
+    - go get -t ./...
```

Comment:
I don't know how the travis `matrix` stuff works, do we get a completely clean environment each time? Do we maybe need to delete the `vendor/` directory from the run above? Or do we need a `-t -u` here to update all the dependencies?

Reply:
The array items in the `include` object overwrite anything defined top-level, but give you a totally separate build for it. You have to copy them exactly into the failure matrix object, in order to have them built, but ignored for overall success.